x-pack/platform/plugins/shared/onechat/server/services/tools/builtin/definitions/dataview_explorer.ts

```diff
@@ -0,0 +1,144 @@
+  return {
+    id: platformCoreTools.dataViewExplorer,
+    type: ToolType.builtin,
+    description: `Find relevant Kibana data views based on a natural language query.
+
+The 'pattern' parameter can be used to filter data views by a specific title pattern, e.g. 'logs*'.
+This should *only* be used if you know what you're doing (e.g. if the user explicitly specified a pattern).
+Otherwise, leave it empty to search against all data views.
+
+*Example:*
+User: "Show me my web server logs"
+You: call tool 'dataview_explorer' with { query: 'data views for web server logs' }
+Tool result: [{ id: 'abc-123', title: 'logs-nginx*', reason: 'Data view for nginx web server logs' }]
+`,
+    schema: dataViewExplorerSchema,
+    handler: async (
+      { query: nlQuery, pattern, limit = 1 },
+      { dataViewsService, modelProvider, logger }
+    ) => {
+      logger.debug(
+        `dataview_explorer tool called with query: ${nlQuery}, pattern: ${pattern}, limit: ${limit}`
+      );
+
+      const model = await modelProvider.getDefaultModel();
+
+      // Get all data views
+      let dataViews = await dataViewsService.getIdsWithTitle();
+
+      // Filter by pattern if provided
+      if (pattern) {
+        const regexPattern = new RegExp(pattern.replace(/\*/g, '.*'));
+        dataViews = dataViews.filter((dv) => regexPattern.test(dv.title));
+      }
+
+      // If no data views, return empty
+      if (dataViews.length === 0) {
+        return {
+          results: [
+            {
+              type: ToolResultType.other,
+              data: {
+                resources: [],
+                message: pattern
+                  ? `No data views found matching pattern: ${pattern}`
+                  : 'No data views found',
+              },
+            },
+          ],
+        };
+      }
+
+      // Use LLM to select most relevant data views
+      const prompt = `Given the following Kibana data views, select the ${limit} most relevant one(s) for this query: "${nlQuery}"
+
+Available data views:
+${dataViews.map((dv) => `- ID: ${dv.id}, Title: ${dv.title}`).join('\n')}
+
+Return ONLY a JSON array of objects with the selected data view IDs and a brief reason why each was selected.
+Format: [{ "id": "data-view-id", "reason": "brief explanation" }]
+
+Select up to ${limit} data view(s).`;
+
+      try {
+        const response = await model.chatModel.invoke([['user', prompt]]);
+        const content =
+          typeof response.content === 'string'
+            ? response.content
+            : JSON.stringify(response.content);
+
+        // Extract JSON from markdown code blocks or plain text
+        const jsonMatch =
+          content.match(/```(?:json)?\s*(\[[\s\S]*?\])\s*```/) || content.match(/(\[[\s\S]*?\])/);
+
+        if (!jsonMatch) {
+          throw new Error('Failed to parse LLM response');
+        }
+
+        const selected = JSON.parse(jsonMatch[1]) as Array<{ id: string; reason: string }>;
```

Comment:
Using structured output would probably be simpler, safer and more efficient here, no? Any reasons you decide to not use it?

I assume you took a look already, but this is how it's done for `index_explorer` - https://github.com/elastic/kibana/blob/1f52a56ad6235dba26678181c843759d75e87891/x-pack/platform/packages/shared/onechat/onechat-genai-utils/tools/index_explorer.ts#L250-L283